### PR TITLE
Port to ocaml

### PR DESCRIPTION
**Pull request by tautologico**
This PR ports the version 2 test runner to OCaml, instead of the current Python version. This has two main benefits:

1. Possible contributors to the OCaml track are more likely to know or want to learn OCaml than another language;
2. OCaml tooling has an easier access to the OCaml compiler libraries, which will make it easier to analyze and manipulate the exercise code 

The Dockerfile was edited to include a build step for the test runner.

### CODE REVIEW

File: bin/run.sh [hunk 1]
```diff
@@ -4,4 +4,4 @@ TOP=$(dirname $( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd
 
 cd "$TOP"
 
-exec ./bin/run.py "${@}"
```
Comment (ErikSchierboom) [inline bin/run.sh:7]: Shouldn't the `run.py` file be removed?

**Review by ErikSchierboom — APPROVED** [review]
Fab work! One tiny nit!

File: bin/run.sh (hunk 1 above)
Comment (georgyo) [reply bin/run.sh:7]: We should also remove the python dependency in the `Dockerfile`

Comment (tautologico) [reply bin/run.sh:7]: I removed the script and the python package installation from the Dockerfile.

**Comment by tautologico** [comment]
Is this ok to be merged? I can't do it.

**Comment by ErikSchierboom** [comment]
Thanks!

**Pull request merged**